### ISSUE TITLE
Weights export 

### ISSUE THREAD

**Issue by AlexRux** [larq/larq#360]
[Sorry, the first time in GitHub]
Hi, I used Netron to see and export the binarized parameters and I see that the parameters aren't. In the attached file you can see what I mean. So, the parameters shouldn't be only +1 and -1? Why is this happening?

The layer: 
```python
kwargs = dict(
    kernel_quantizer=lq.quantizers.SteSign(clip_value=1.0),
    kernel_constraint="weight_clip",
    padding="same"
)

activation = 'ste_sign'

x = tf.keras.Input(shape=(1, Time_series_length, 1))
ConvL1 = lq.layers.QuantConv2D(128, (1, 8), use_bias=False, **kwargs)(x)
ConvL1 = tf.keras.layers.BatchNormalization(scale=False)(ConvL1)
ConvL1 = tf.keras.layers.Activation(activation)(ConvL1)
```

![image](https://user-images.githubusercontent.com/45176956/71084726-35399a80-2196-11ea-81a6-6616009a064b.png)

In the image, you can see the parameters of the layer. 

TensorFlow version: 2.0.0
Larq version: 0.8.2

**Comment by lgeiger** [comment]
> [Sorry, the first time in GitHub]

No worries, thanks for the issue :+1:

> So, the parameters shouldn't be only +1 and -1? Why is this happening?

Unfortunately this is currently expected. Since you are using `lq.quantizers.SteSign(clip_value=1.0)` to quantized the underlying [latent weights](https://larq.dev/guides/bnn-optimization/#latent-weights) are only binarized in the forward pass and the weights stored in TensorFlow are not actually binary unless you use an optimizer like [Bop](https://larq.dev/api/optimizers#bop). We don't save the binary versions of the weights by default in order to support correct resuming during training when using model checkpoints to restore.

With #312 merged we will now be able to support saving of quantized versions of the weights, but it might take a bit until we figure out a proper public API for that. We are also currently working on a TFLite based implementation for inference that will take care of this correctly by relying on TensorFlows constant folding optimization, but this will take a bit longer to be ready to use.

**Comment by XAXOXAX** [comment]
> Unfortunately this is currently expected. Since you are using `lq.quantizers.SteSign(clip_value=1.0)` to quantized the underlying [latent weights](https://larq.dev/guides/bnn-optimization/#latent-weights) are only binarized in the forward pass and the weights stored in TensorFlow are not actually binary unless you use an optimizer like [Bop](https://larq.dev/api/optimizers#bop). 

So the accuracy takes account of the binarized ones, right? But we can't export them right now.  

>We don't save the binary versions of the weights by default in order to support correct resuming during training when using model checkpoints to restore.

This makes sense. My next step should be to implement the network on a microcontroller, so I need the raw version of the binarized weights to be uploaded on the machine.

**Comment by leonoverweel** [comment]
> So the accuracy takes account of the binarized ones, right?

Yep, the [`ste_sign` quantizer](https://larq.dev/api/quantizers/#ste_sign) binarizes the weight when doing inference, so the accuracy is calculated as if the weight was binary.

> But we can't export them right now.

> My next step should be to implement the network on a microcontroller, so I need the raw version of the binarized weights to be uploaded on the machine.

We can't export them explicitly right now, but the `ste_sign` is quite simple: any latent weight (the numbers you screenshotted from Netron) that is < 0 becomes -1 and any >= 0 becomes +1 ([see docs]((https://larq.dev/api/quantizers/#ste_sign))). For now, until our TFLite inference implementation is ready, you could apply this quantization rule yourself when you're uploading the weights to the machine.

**Comment by lgeiger** [comment]
> My next step should be to implement the network on a microcontroller, so I need the raw version of the binarized weights to be uploaded on the machine.

As @leonoverweel mentioned, there is no nice way to do this currently. There are two ways you could workaround this for now:
1. Use an optimizer like [Bop](https://larq.dev/api/optimizers#bop) which doesn't rely on latent weights
2. Install `larq` from GitHub `master` and use the experimental `quantization_scope` API. Note that the details might change before the 0.9 release, but I'll update this issue if the API changes.
```python
import tensorflow as tf
import larq as lq
from larq import quantized_scope

model = tf.keras.models.Sequential(
    [
        lq.layers.QuantConv2D(
            64, 3, kernel_quantizer="ste_sign", padding="same", input_shape=(32, 32, 3)
        ),
        lq.layers.QuantConv2D(
            64,
            3,
            input_quantizer="ste_sign",
            kernel_quantizer="ste_sign",
            activation="relu",
        ),
        tf.keras.layers.Flatten(),
        tf.keras.layers.Dense(10, activation="softmax"),
    ]
)

model.save("full_precision.h5")

with quantized_scope.scope(True):
    model.save("binary.h5")
```

**Comment by AlexRux** [comment]
> For now, until our TFLite inference implementation is ready, you could apply this quantization rule yourself as preprocessing when you're uploading the weights to the machine.

Oh, perfect. A Matlab script and I'm ready to go. 


> 1. Use an optimizer like [Bop](https://larq.dev/api/optimizers#bop) which doesn't rely on latent weights
> 2. Install `larq` from GitHub `master` and use the experimental `quantization_scope` API. Note that the details might change before the 0.9 release, but I'll update this issue if the API changes.

Thank you both. I'll try.

**Comment by lgeiger** [comment]
Great! Feel free to reopen if you have more questions.
